kubeflow/fairing/cloud/gcp.py

```diff
@@ -64,11 +64,17 @@ def add_gcp_credentials_if_exists(kube_manager, pod_spec, namespace):
         else:
             logger.warning("Not able to find gcp credentials secret: {}"\
                            .format(constants.GCP_CREDS_SECRET_NAME))
+            logger.warning("Trying workload identity service account: {}"\
+                           .format(constants.GCP_SERVICE_ACCOUNT_NAME))
+            pod_spec.service_account_name = constants.GCP_SERVICE_ACCOUNT_NAME
     except Exception as e:
         logger.warning("could not check for secret: {}".format(e))
 
 
 def add_gcp_credentials(kube_manager, pod_spec, namespace):
+    """Note: This method will be deprecated soon and will become unavailable by 1.0.
```

Comment:
use https://docs.python.org/3/library/exceptions.html#DeprecationWarning ? e.g. https://stackoverflow.com/questions/9008444/how-to-warn-about-class-name-deprecation

Reply:
Given that this function is only called by add_gcp_credentials_if_exists and is not directly used by any job configurator, not sure if all of that warning mechanism makes sense. Essentially, add_gcp_credentials_if_exists will become just adding the service_acccount_name to the pod_spec template.